nachetDb-1.0.0/db-creation.py

```diff
@@ -18,34 +18,34 @@
 #     )
 # """ % ("nachetdb_1.0.0"))
 
-# # Create Indexes table
-# cur.execute("""
-#     CREATE TABLE  \"%s\".indexes (
-#         id SERIAL PRIMARY KEY,
-#         index JSON,
-#         ownerID uuid REFERENCES "nachetdb_1.0.0".users(id)
-#     )
-# """ % ("nachetdb_1.0.0"))
+# Create Indexes table
+cur.execute("""
+    CREATE TABLE  \"%s\".indexes (
+        id uuid  PRIMARY KEY,
+        index JSON,
+        ownerID uuid REFERENCES "nachetdb_1.0.0".users(id)
+    )
+""" % ("nachetdb_1.0.0"))
 
-# # Create Pictures table
-# cur.execute("""
-#     CREATE TABLE  \"%s\".pictures (
-#         id SERIAL PRIMARY KEY,
-#         picture JSON,
-#         indexID INTEGER REFERENCES "nachetdb_1.0.0".indexes(id)
-#     )
-# """ % ("nachetdb_1.0.0"))
+# Create Pictures table
+cur.execute("""
+    CREATE TABLE  \"%s\".pictures (
+        id uuid  PRIMARY KEY,
+        picture JSON,
+        indexID uuid REFERENCES "nachetdb_1.0.0".indexes(id)
+    )
+""" % ("nachetdb_1.0.0"))
```

Comment:
don't get into the habit of bypassing the built in parameters facilities

https://www.psycopg.org/psycopg3/docs/basic/params.html

If the variables containing the data to send to the database come from an untrusted source (such as data coming from a form on a web site) an attacker could easily craft a malformed string, either gaining access to unauthorized data or performing destructive operations on the database. This form of attack is called [SQL injection](https://en.wikipedia.org/wiki/SQL_injection) and is known to be one of the most widespread forms of attack on database systems. Before continuing, please print [this page](https://xkcd.com/327/) as a memo and hang it onto your desk.

Reply:
You cant use parameters query for schema, table or column name. I understand the need to protect from injection and will apply the parameters query for my future cursor.

Reply:
As discussed yesterday, use ```set search_path``` for DDL operations

https://www.postgresql.org/docs/current/ddl-schemas.html

>Qualified names are tedious to write, and it's often best not to wire a particular schema name into applications anyway. Therefore tables are often referred to by unqualified names, which consist of just the table name. The system determines which table is meant by following a search path, which is a list of schemas to look in. The first matching table in the search path is taken to be the one wanted. If there is no match in the search path, an error is reported, even if matching table names exist in other schemas in the database.